src/components/AttachmentModal.js

```diff
@@ -330,7 +337,13 @@ function AttachmentModal(props) {
                     setShouldLoadAttachment(true);
                 }}
                 onModalHide={(e) => {
-                    props.onModalHide(e);
+                    props.onModalHide(e).then(() => {
+                        if (!onModalHideCallbackRef.current) {
```

Comment:
I would check also if it's a function, because later we call it.

Additionally, I think it would be nice to extract this event handler to a function.

Reply:
Ok, I see there was a check if it's a function previously - why it's removed?

Reply:
It came from this comment. Basically it can only be null or a function